python/meson.build

```diff
@@ -0,0 +1,18 @@
+project(
+    'nanoarrow',
+    'cpp', 'cython',
+    version: '0.6.0.dev0',  # TODO: do we need git archive support?
+    license: 'Apache-2.0',
+    meson_version: '>=1.2.0',
+    default_options: [
+        'warning_level=2',
+        'cpp_std=c++17',
+        'default_library=static',
+        'nanoarrow:ipc=true', # https://github.com/mesonbuild/meson/issues/6728
```

Comment:
Ideally we wouldn't need both of these. However, in the current configuration `arrow-nanoarrow` is assumed to be a symlink whereas `nanoarrow` is assumed to be a downloaded external dependency.

In theory we could just change `nanoarrow.wrap` to `arrow-nanoarrow.wrap` and let Meson do its magic, consolidating these options into 1. The problem though, is that the bootstrap.py logic needs to look in the subprojects folder to copy some files, and is unclear when you have multiple arrow-nanoarrow subproject directories which one the bootstrap.py script should reference